### PR TITLE
Re-enable builds under macOS 10.12.

### PR DESCRIPTION
**Pull request by droark**
Once again, Apple breaks stuff, and we have to un-break it.

### CODE REVIEW

**Comment by droark** [comment]
I may need to add a README_OSX file or something similar stating that users must now use `brew` to install OpenSSL. Long story short, without the brewed OpenSSL, running Armory throws up scary warnings on the command line regarding HTTPS being unreliable and subject to MitM attacks. This is traced back to Python, which, for whatever reasons, can't cope with the ancient version of OpenSSL that Apple ships with OS X. The only solution is to download a standardized version that can also be used to properly compile Python. Ergo, `brew` comes into play (along with a couple of new but small Python packages). Not ideal but I've seen far worse.

File: README_OSX.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# macOS (OS X) README
+## Requirements
+As of Armory 0.95.1, due to Python requirements that can't be met by macOS 10.12 (Sierra), users must use `brew` to install an updated version of OpenSSL in order for Armory to run. Please follow these instructions.
+
+1. Open a terminal and run the `brew` command. If `brew` is not found, please execute the following commands from the command line. If the final command returns any errors, consult Google, the *bitcoin-armory* IRC channel on Freenode, or the [Bitcoin Forum](https://bitcointalk.org/index.php?board=97.0) for further instructions.
+
+        /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+        touch ~/.bashrc
+        echo "export CFLAGS=\"-arch x86_64\"" >> ~/.bashrc
```
Comment (droark) [inline README_OSX.md]: Good question. I'm going off [directions](https://gist.github.com/bsmt/starred) that seem to have been copied & pasted throughout the Internet. I'm guessing the .bashrc lines are meant to ensure that a 32-bit version isn't built by accident.

**Comment by droark** [comment]
Hello. Just FYI, I *think* the only place where Armory uses OpenSSL (via Python) is in the hashing functions. If you uninstall the `brew`ed version of OpenSSL and try to run the binary, you see complaints such as the following.

```
ERROR:root:code for hash md5 was not found.
Traceback (most recent call last):
  File "/Users/droark/Projects/private-goatpig-BitcoinArmory/osxbuild/workspace/Armory.app/Contents/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 147, in <module>
  File "/Users/droark/Projects/private-goatpig-BitcoinArmory/osxbuild/workspace/Armory.app/Contents/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 97, in __get_builtin_constructor
ValueError: unsupported hash type md5
```

The hashes (MD5, SHA1, and the SHA2 variants) seem to be all that Python cares about. So, if Armory uses Python to hash anything (I'm pretty sure it does), I believe this means that OpenSSL is used in the background. OpenSSL appears to also be used if any HTTPS links are followed. So, for better or worse, it'll be important for OS X users to keep their `brew`ed OpenSSL binaries up to date if there are any relevant security bugs found. (Granted, I'm pretty sure Linux and Windows are in the same boat.) I'll add that, because a bunch of people asked at the time, this doesn't apply to things like the Heartbleed vulnerability.

**Comment by goatpig** [comment]
There is no more phone home code in Armory and I don't intent to add any in the future. If I move all hashing from Python's own code (which itself depends on OpenSSL on Macs only apparently?) to the Swig'd CryptoPP method, would it remove the dependency entirely?

**Comment by goatpig** [comment]
Also your PR refers users to the latest version of XCode. Is XCode 8 now functional?

**Comment by droark** [comment]
@goatpig - Actually, I was wrong. It's Twisted that cares. I'm not exactly sure what's going on there. Here's one of the messages that comes up.

```
You do not have a working installation of the service_identity module: 'No module named pyasn1.codec.der.decoder'.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module and a recent enough pyOpenSSL to support it, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
```

Even if Armory isn't actually following any links, this is the kind of thing that could cause somebody to freak out if they notice it. (Not that I expect many Mac users to run Armory from the command line. Anyway....) I'd rather nip it in the bud than let it sit there and wait for somebody to find it and post on Reddit or wherever.

Also, yes, I'm running XCode 8.1 now. It works fine with this fix. (I can't speak to XCode 8.0 but I don't care. :) )

**Comment by droark** [comment]
Also, in case anyone stumbles upon this but missed the IRC talk from a couple of days ago, I originally wanted to drop OS X 10.7 support alongside this PR. Its support of C++11 is a bit half-baked, to the point that [Core dropped 10.7 as of 0.13.1](https://github.com/bitcoin/bitcoin/pull/9022). This could be a problem as Armory moves more and more towards C++. After talking to goatpig, I've decided to wait 'til at least 0.96 or thereabouts so that, if anyone is using 10.7 for offline signing with SegWit and/or the new wallet (and they're not bit by some obscure C++ bug), they'll be covered.

**Comment by droark** [comment]
Weird. I just pulled my fix into the latest version of dev. (I was using the one from when 0.95 went out.) Turns out I was wrong about this fixing the XCode 8 issue. :/ Looks like you either have to [force a 10.8 minimum runtime](https://forums.developer.apple.com/thread/63266) or build using XCode 7 on 10.11. I'll update the build notes. In any event, I'm not making any promises if someone comes out of the woodwork and says Armory doesn't work on 10.7!

**Comment by goatpig** [comment]
I'm not either. Once I move the RPC functionality out of the client, I can get rid of the Python twisted dependency. We'll reassess the build process then.

**Pull request merged**